### PR TITLE
Fix DustAppRun, allow local run in dev

### PR DESCRIPTION
**Pull request by spolu**
- Fixes DustAppRun end-to-end (UI still needs to be done)
- Allow running a DustApp locally in dev

Deployment:
- init_db on front
- deploy front

### CODE REVIEW

**Comment by spolu** [comment]
Post-merge review welcome

File: front/lib/api/assistant/actions/dust_app_run.ts [hunk 1]
```diff
@@ -357,8 +361,12 @@ export async function* runDustApp(
   // Let's run the app now.
   const now = Date.now();
 
-  const prodCredentials = await prodAPICredentialsForOwner(owner);
-  const api = new DustAPI(prodCredentials);
+  const prodCredentials = await prodAPICredentialsForOwner(owner, {
```
Comment (fontanierh) [inline front/lib/api/assistant/actions/dust_app_run.ts:364]: I feel this is becoming confusing. Why is it "prod api credentials" then ? if we're already in prod, then it's just "regular" credentials, and if we're not in prod and it's using local, then it's not prod credentials ?

File: front/lib/api/assistant/actions/dust_app_run.ts [hunk 2]
```diff
@@ -357,8 +361,12 @@ export async function* runDustApp(
...
+    useLocalInDev: true,
```
Comment (fontanierh) [inline front/lib/api/assistant/actions/dust_app_run.ts:365]: Should this be in an env var ?

File: front/lib/api/assistant/agent.ts [hunk 3]
```diff
@@ -323,9 +323,9 @@ export async function* runAgent(
         }
       }
     } else {
-      throw new Error(
-        "runAgent implementation missing for action configuration"
-      );
+      ((a: never) => {
```
Comment (fontanierh) [inline front/lib/api/assistant/agent.ts:326]: 🙏

**Review by fontanierh — COMMENTED** [review]
✨

File: front/lib/api/assistant/actions/dust_app_run.ts (hunk 1 above)
Comment (spolu) [reply front/lib/api/assistant/actions/dust_app_run.ts:364]: This is always prod unless that new special flag

File: front/lib/api/assistant/actions/dust_app_run.ts (hunk 2 above)
Comment (spolu) [reply front/lib/api/assistant/actions/dust_app_run.ts:365]: No because we want it for the app but not the actions

**Pull request merged**